components/motor/gpio/motor_encoder.go

```diff
@@ -464,6 +481,11 @@ func (m *EncodedMotor) GoFor(ctx context.Context, rpm, revolutions float64, extr
 	if rpm == 0 {
 		return motor.NewZeroRPMError()
 	}
+
+	rpm *= float64(m.flip)
+
+	m.state.lastPowerPct = 0 // set power to zero when GoFor is called
```

Comment:
The "when GoFor is called" seems redundant, as we already know we're at the top of that function. How about "stop any previous movement" or something like that instead?

Reply:
Mmm, this is more to address the bug that lastState was being stored to a non zero but small value other wise. So this is clearing that value to zero to start calculating power from a zero state again. 
I can say `ensure power to zero at the start of a call` if that's more appropriate

Reply:
The favorite book I brought in to the office is _The Art of Readable Code_ by Boswell and Foucher*. Chapter 5 is all about knowing what to comment, and one of their main ideas is that you shouldn't comment things that can easily be gotten from the code itself. I had objected to the "when GoFor is called" because the reader already knows this is at the top of GoFor, but more broadly, commenting that we set the value to 0 isn't so great because the reader can easily see that in the code directly, too. and replacing "when GoFor is called" with "at the start of a call" doesn't add anything, and replacing "set power to zero" with "ensure power to zero" also doesn't add anything.

The important thing is _why_ we're setting it to 0, and if there should be a comment at all, it should be about that instead. Why is it a problem if we store a small-but-nonzero value at the beginning?


* Full disclosure: the _AoRC_ authors are friends of mine, I was a proofreader before the book was published, and I'm thanked in the preface. I suspect I'd still think very highly of this book if I wasn't, but I'm somewhat biased.

Reply:
Ensuring that the saved variable is  zero at the start of the call is what's important. We want to save the power state for our calculations, but we want to clear the variable when we start another request. This is common motor/controls logic, not sure how to describe it better.

Reply:
In that case I'd get rid of the comment entirely. The code by itself says we set this to 0 at the start of GoFor, and duplicating that in a comment will only slow people down.

Reply:
I actually think adding teaching comments is important, especially since a lot of our engineers are not robotics engineers and don't think of this pattern, but software engineers learning robotics. So vetoing removing it, but open to a better comment describing why clearing the state at the beginning of this logic is important.

Reply:
I do like that last sentence: if the comment had been "clear any leftover state from last time," that adds something that could not be gotten straight from the code itself.